### ISSUE TITLE
Missing type argument of generic extension method is not detected

### ISSUE THREAD

**Issue by seritools** [dotnet/roslyn#16101]
**Version Used**: 
VS2015 Update 3

**Steps to Reproduce**:

1. Quick console application code to reproduce the error:
```csharp
namespace RoslynBugTest
{
    class Program
    {
        static void Main(string[] args)
        {
            string works1 = RoslynBugTest.TestStatic<string>();
            string fails1 = RoslynBugTest.TestStatic<>(); // ERROR

            var testClass = new RoslynBugTest();

            string works2 = testClass.TestMethod<string>();
            string fails2 = testClass.TestMethod<>(); // ERROR

            string works3 = testClass.TestExtensionMethod<string>();
            string shouldFail = testClass.TestExtensionMethod<>(); // NO ERROR
        }
    }

    class RoslynBugTest
    {
        public static T TestStatic<T>() => default(T);
        public T TestMethod<T>() => default(T);
    }

    static class RoslynBugTestExtensions
    {
        public static T TestExtensionMethod<T>(this RoslynBugTest r) => default(T);
    }
}
```
2. Observe that VS does not show any error for shouldFail3
3. Compile the code while commenting out the declaration/assignments of fails1 and fails2

**Expected Behavior**:
Squiggly red line and the following error message:
`CS0305: Using the generic method group 'TestExtensionMethod' requires 1 type arguments`

**Actual Behavior**:
No message, squiggly line or anything in VS while coding, generic error while compiling (when fails1 and fails2 are commented out):
`CS7038: Failed to emit module 'RoslynBugTest'.`
No hints as to which file/line/code failed to compile.

**Comment by SLaks** [comment]
This is already fixed on the master branch.

http://tryroslyn.azurewebsites.net/#b:master/f:r/AQ4YwGwQwZx4AKAnA9gcyVAtgKFMAbz31BgBcoyBLMYANxSoBNgBZKKgOwApyku0AbQC6wKEjQwAlMRKFZc0mX6c0wAO4okAaxgBGYAF5gAJRQwIAT04AhAK5oAKgFNyAOhfkAyhWpgAPHwCAHzcUgDcCorAAPQxwEGqwABmHBD6RqbmVrYOnmQermQ+lDT+oRGx8QCiJiYA8iY4UYp04sBkRQDC0HCZnM7qWRbW9k5FYZEtcolqmjowAEyZneQ9sDCF5KzOZAAWKEyByiGT0yRxsylpSyvdvZv5O/uH5ZNVwLUNTef4V/O6ADMdzWDy2ZGqAA9OpwYFQUJxngcjrMKpFon8TkkYAc7BAmAAxNIgsjrODgqEwuEIpGvNEfABy9U+dUa5wAvgpOQpIBthjkxvkFERogAHOwAIwgNASvhljmA+RKfn8jgqRmCwCYzlSeLI3EcEXO4ql8sVRVpRzVYQ1Wp1UD1BqN+G5mNKtF5fTMI1y43IlOcsPhsOFxsl0to5HdivN/uhgepiN2yNVoX2VHg3oFeSKwCQUlt2t1EH1hvRoE5QA===

Changing it to Release shows the bug.

**Comment by seritools** [comment]
Thanks for the quick response! Didn't know about the nice tryroslyn site!
